File: templates/hexagonal-spring-boot/domain/src/main/java/packageName/domain/ExampleDomain.java
package packageName.domain;

import packageName.domain.model.ExampleInfo;
import packageName.domain.port.ObtainExample;
import packageName.domain.port.RequestExample;

public class ExampleDomain implements RequestExample {

  private ObtainExample obtainExample;

  public ExampleDomain() {
    this(new ObtainExample() {
    });
  }

  public ExampleDomain(ObtainExample obtainExample) {
    this.obtainExample = obtainExample;
  }

  @Override
  public ExampleInfo getExamples() {
    return ExampleInfo.builder().examples(obtainExample.getAllExamples()).build();
  }
}

File: templates/hexagonal-spring-boot-kotlin/domain/src/main/kotlin/packageName/domain/ExampleDomain.kt
package packageName.domain

import packageName.domain.model.ExampleInfo
import packageName.domain.port.ObtainExample
import packageName.domain.port.RequestExample

class ExampleDomain(private val obtainExample: ObtainExample) : RequestExample {

    constructor() : this(object : ObtainExample {})

    override fun getExamples(): ExampleInfo {
        return ExampleInfo(obtainExample.getAllExamples())
    }
}
